### PR TITLE
Simplify & shorten posixlib netdb code paths

### PR DESCRIPTION
**Pull request by LeeTibbert**
This PR extends the work done in pending PR https://github.com/scala-native/scala-native/pull/2742.

Code in recently merged PRs allows an additional number of netdb related code paths
to be simplified and shortened. Files which are now unnecessary/redundant are removed.
This should bot improve maintaining correctness and reduce execution time.  Network
name lookups are expensive at best and are frequently done, usually when somebody
is waiting.  Milliseconds gained are worthwhile, as long as correctness prevails.

The number of Windows include files could possibly be reduced/minimized by
someone who knows Windows and has access to a Windows development
system.

### CODE REVIEW

File: posixlib/src/main/scala/scala/scalanative/posix/netdb.scala [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
+   * the FreeBSD layout.
+   *
+   * Handled in netdbOps below.
+   */
+
+  /* _Static_assert code in netdb.c checks that Scala Native and operating
+   * system structute definitions match "close enough". If you change
+   * something here, please make the corresponding changes there.
+   */
+
```
Comment (ekrich) [inline posixlib/src/main/scala/scala/scalanative/posix/netdb.scala:24]: These should probably be a formal Scaladoc - we do generate docs for these.  Int the C lib you can see them - https://javadoc.io/doc/org.scala-native/clib_native0.4_2.13/latest/scala/scalanative/libc/fenv$.html and also the next two. Complex has a comment on top.

The links to docs are on the main github page.

Comment (LeeTibbert) [reply posixlib/src/main/scala/scala/scalanative/posix/netdb.scala]: Thank you for the timely review.

Eh?  Can you educate me on how it is created?  
Sorry but I do not know from Scaladoc.  

On a quick look, Scaladoc looks like Huge overhead for something that
is well described, modulo OS differences, on 'man' available on any system.

Camel can carry only so many bales.
I thought I was doing well writing it down at least once.

Comment (LeeTibbert) [reply posixlib/src/main/scala/scala/scalanative/posix/netdb.scala]: I took a quick look at Scaladoc conventions. I am not adverse to someone doing
a PR after this one is merged to add Scaladoc but feel that lack of Scaladoc should
not hold up this PR.

Comment (ekrich) [reply posixlib/src/main/scala/scala/scalanative/posix/netdb.scala]: As just a note with no fancy stuff would suffice for what you have as a start. At the top of the `class`, `object` or just before any `type` or `def` you can put a comment.
```Scala
/** This is netdb that does xxx
 *
 *  Another paragraph saying how great it is
 */
object netdb { ...
```
Scalafmt will format it correctly.

**Comment by LeeTibbert** [comment]
All green, with the exception of a known & entirely unrelated failure in one qemu.
Ready for review.

Probably easier if this is merged after related #2742. Less chance of
needing a git rebase that way.

File: posixlib/src/main/scala/scala/scalanative/posix/netdb.scala (hunk 1 above)
Comment (LeeTibbert) [reply posixlib/src/main/scala/scala/scalanative/posix/netdb.scala]: Trying to be a good kid, I checked the Contributor's Guide and noticed the total
absence of a section on Scaladoc and how to generate it within the Scala Native
build.   

Not to give in to a bad attitude, I figured out how to generate Scaladoc within
the Scala Native build. Old news to some, but not evident (sbt posixlib3/doc).
Posixlib was all that I needed, so I stopped before figuring out how to generate
the Scaladoc for all projects. 

Then I figured out how to view the local copy of the generated Scaladoc (location
and name of generated html files).  

What that showed me is a listing of objects, for each object there were methods and
types.  I checked this netdb.scala and a couple of others.  The descriptions
could certainly be improved by careful work, but they are, IMO, passable.

So, I believe that minimal Scaladoc comes for 'free' and ticks the
"is Scaladoc present" checkbox.  

Adding the items Eric suggests should be considered as desirable, but, again IMO, 
should not be a hard requirement.  Especially in the light of prior art and the total
lack of information in the Contributor's Guide.

**Comment by LeeTibbert** [comment]
Just added two sections of Scaladoc, one for each object.  The `netdb` object gives a URL
to the Open Group netdb.h specification and specification year.   This gives
anyone reading either the file or the Scaladoc exactly which specification & year
we are trying to follow.

The text can be wordsmithed & improved.  Best I can do at this hour
of the night.

I have a  private message out to Eric to see if we can do some iterative
improvement as files are touched in the future.

**Comment by LeeTibbert** [comment]
Rebased & all but usual multiarch suspects pass CI.
Multiarch have usual failures apparently un-related to this PR.

**Pull request merged**